src/login/org.freedesktop.login1.policy.in

```diff
@@ -218,6 +218,39 @@
                 <annotate key="org.freedesktop.policykit.imply">org.freedesktop.login1.reboot</annotate>
         </action>
 
+        <action id="org.freedesktop.login1.halt">
+                <_description>Halt the system</_description>
+                <_message>Authentication is required for halting the system.</_message>
+                <defaults>
+                        <allow_any>auth_admin_keep</allow_any>
+                        <allow_inactive>auth_admin_keep</allow_inactive>
+                        <allow_active>yes</allow_active>
```

Comment:
Should we be paranoid and make these all `auth_admin_keep`?

When you look up what `halt` does, people talk about returning to a ROM monitor.  I don't know whether there are such ROM monitors that might include a boot password, and whether they'd always be expected to re-prompt for the boot password when the launched software returns to them.

It's not ideal - `systemctl halt` returning EACCESS when the user might benefit from a hint to `systemctl poweroff` instead.  But I don't think it's different from what we do at the moment.  And hopefully it is a marginal issue, with existing users having already been re-educated.

Reply:
true, it makes sense protecting this more than the other verbs, given that "halt" is pretty low-level and has this "non-recoverable dead-end feeling" to it. will change.